### PR TITLE
Assignment4.js

### PR DESCRIPTION
**Pull request by TlucasX5**
Chapter 3 assignments

### CODE REVIEW

File: assignments/Assignment4.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+max3(3.14, 3.141, -3.145)
+
+
+
+// #5 Sum of odd integers p.112
+function sumOfOddIntegers(n) {
+    let sum = 0;
+    let ctr = 1;
+    // Start from 1 and increment by 2 fro odd numbers
+    for(let i = 1; ctr <= n; i += 2) {
```
Comment (akennis) [inline assignments/Assignment4.js:24]: This is correct but it is very (very) unusual to not use the same variable in all three parts of the `for` loop - in this case that would be `i`.

Also, `i` is almost always used as a simple index or counter variable and any other state you need to maintain should be in a separate, appropriately named variable.

You can probably get away with having only two variables in fact, `i` for the simple counter and `sum` as the aggregator.  If I asked you to just tell me what the nth odd number is, can you find that based on a simple formula?

File: assignments/Assignment4.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+
+// #2 Return the largest of 3 arguments numeric or string p.112
+function max3(x, y, z) {
+    if ((x > y) && (x > z)) {
+        return console.log(x + " is the largest");
```
Comment (akennis) [inline assignments/Assignment4.js:5]: You should be returning the largest value itself.  This code currently returns the result of calling `console.log` which is undefined (i.e. `console.log` doesn't itself return a value).

File: assignments/Assignment4.js (hunk 1 above)
Comment (TlucasX5) [reply assignments/Assignment4.js:24]: I tried to do that but  I couldn't get it to work correctly until I broke out a separate counter that tracked the iterations of the number initially entered (N = 4) and the i ctr kept track of the odd numbers needed to be added together. For example, N= 4 equals 1 + 3+ 5+7 = 16. When I enter 5 = result is 25. 
function sumOfOddIntegers(n) {
    let sum = 0;
    let ctr = 1;
    // Start from 1 and increment by 2 fro odd numbers
    for(let i = 1; ctr <= n; i += 2) {
        sum = sum + i;
        ctr = ctr + 1;
    }
    return sum;
  
  }

  console.log("The sum of odd numbers for this integer is: " + sumOfOddIntegers(4));
  console.log("The sum of odd numbers for this integer is: " + sumOfOddIntegers(5));
  console.log("The sum of odd numbers for this integer is: " + sumOfOddIntegers(6));

RESULT:
The sum of odd numbers for this integer is: 16
The sum of odd numbers for this integer is: 25
The sum of odd numbers for this integer is: 36

File: assignments/Assignment4.js (hunk 2 above)
Comment (TlucasX5) [reply assignments/Assignment4.js:5]: It is working when I run it in the browser. I was googling and came across this but from your lecture tonight your saying console.log should not be returned for results. So I should have returned the value then console.log to print.
function max3(x, y, z) {
    if ((x > y) && (x > z)) {
        return console.log(x + " is the largest");
    } else if ((y > x) && (y > z)) {
        return console.log(y + " is the largest");
    } else {
        return console.log(z + " is the largest");
    }
}

max3(30, 50, 20)
max3("bite", "bat", "bit")
max3(3.14, 3.141, -3.145)

50 is the largest
bite is the largest
3.141 is the largest

Comment (TlucasX5) [reply assignments/Assignment4.js:5]: // #2 Return the largest of 3 arguments numeric or string p.112
function max3(x, y, z) {
    if ((x > y) && (x > z)) {
        return x;
    } else if ((y > x) && (y > z)) {
        return y;
    } else {
        return z;
    }
}
console.log(max3(30, 50, 20));
console.log(max3("bite", "bat", "bit"));
console.log(max3(3.14, 3.141, -3.145));

**Pull request closed without merging**